apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PatientClaimsEobTaskTransformer.java

```diff
@@ -0,0 +1,374 @@
+ * non-singleton spring-managed bean; each task requires its own unique instance of a {@link
+ * EntityManager} since processing requires fetching data from a database and requires its own db
+ * connection to operate independently of other concurrent claim tasks. All BFD db connections are
+ * injected by a Spring-managed {@link com.zaxxer.hikari.HikariDataSource}.
+ *
+ * <p>To achieve a unique {@link EntityManager}, this class defines its scope as "prototype", which
+ * identifies its need have its own unique instance of the {@link EntityManager}.
+ *
+ * <p>Once Spring has created the class, the caller then sets up the task parameters that define the
+ * operating constraints {@link PatientClaimsEobTaskTransformerV2#setupTaskParams}. In addition,
+ * some claim tasks require an additional property denoting how NPI tax number is processed; this
+ * property is set using the {@link PatientClaimsEobTaskTransformerV2#setIncludeTaxNumbers}; the
+ * default property value for inclusion of NPI tax info is FALSE (do not include NPI tax info).
+ */
+@Component
+@Scope("prototype")
+public class PatientClaimsEobTaskTransformer implements Callable {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(PatientClaimsEobTaskTransformer.class);
+
+  // +++++++++++++++++++++++++++++++++++
+  // bean injected parameter values
+  // +++++++++++++++++++++++++++++++++++
+
+  /** capture performance metrics. */
+  private final MetricRegistry metricRegistry;
+  /** drug description lookup table. */
+  private final FdaDrugCodeDisplayLookup drugCodeDisplayLookup;
+  /** NPI lookup table. */
+  private final NPIOrgLookup npiOrgLookup;
+  /** The samhsa matcher. */
+  private final Stu3EobSamhsaMatcher samhsaMatcher;
+  /** Database entity manager. */
+  private EntityManager entityManager;
+
+  // +++++++++++++++++++++++++++++++++++
+  // setup parameter values
+  // +++++++++++++++++++++++++++++++++++
+
+  /** the ClaimTransformer interface for execution. */
+  private ClaimTransformerInterface claimTransformer;
+  /** the claim type to retreive data for. */
+  private ClaimType claimType;
+  /** beneficiary identifier to process. */
+  private Long id;
+  /** date range that lastUpdate falls within. */
+  private Optional<DateRangeParam> lastUpdated;
+  /** date range that clm_thru_dt falls within. */
+  private Optional<DateRangeParam> serviceDate;
+  /** whether to return tax numbers. */
+  private boolean includeTaxNumbers = false;
+  /** whether to exclude SAMHSA claims. */
+  private boolean excludeSamhsa;
+
+  // +++++++++++++++++++++++++++++++++++
+  // task properties
+  // +++++++++++++++++++++++++++++++++++
+
+  /** capture exception if thrown. */
+  private Exception taskException = null;
+  /** keep track of EOBs that were not removed (ignored) by SAMHSA iterations. */
+  private int samhsaIgnoredCount = -1;
+  /** keep track of SAMHSA removals. */
+  private int samhsaRemovedCount = 0;
```

Comment:
An object called from multiple threads should either be immutable, synchronize access to its mutable fields, or declare those fields as volatile.

Any of these that don't have to be fields should be locals.  Others should be volatile or have synchronization on their reads and writes.

Reply:
I think each of these classes are created with their own thread (this isnt a singleton instance) so I think these are ok; I dont _think_ they are ever going to be called from multiple threads.

Reply:
They are created on one thread then executed in a thread pool and then the results are retrieved from the original thread.

Reply:
so used AtomicInteger for the counters; declared the Collection<EOB> as final; wrapped setter for storing exception info with concurrent synchronized. So while the original premize of running in its own memory space, changes should make any possible threading funkiness a bit more bullet-proof.